packages/models/src/codeObjectId.js

```diff
@@ -1,8 +1,12 @@
 import { CodeObjectType } from './codeObjectType';
 
 export default function codeObjectId(codeObject, tokens = []) {
-  if (codeObject.parent) {
-    codeObjectId(codeObject.parent, tokens);
+  const { parent } = codeObject;
+
+  // If it's a route or query we don't need to include the parent name because it's always the same
+  // ('HTTP server requests' for route and 'Database' for queries). This mirrors the VS Code implementation.
+  if (parent && parent.type !== CodeObjectType.HTTP && parent.type !== CodeObjectType.DATABASE) {
```

Comment:
External services as well. The most accurate definition would be any parent that's not a package or class.

Reply:
I tried changing to definition to be any parent that's not a package or class and it made [this unit test](https://github.com/getappmap/appmap-js/blob/main/packages/models/tests/unit/codeObjectId.spec.js#L85-L95) fail, so I specifically checked for queries, routes, and external services.

Reply:
I'd argue the assertion made in that unit test is invalid and the VS Code implementation is also wrong by including anything separated by `->`. I don't believe those separators were intended to make their way into the FQID.

In any case, this implementation is at least consistent with VS Code now.